From 581cfa1ffdf3f0beabe7734ea90ffe1f29347fd1 Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Mon, 2 Feb 2015 17:08:46 -0600
Subject: [PATCH 03/13] remoteproc/pruss: fix the cleanup path in
 pru_rproc_probe()

The PRU remoteproc driver can manually boot a processor if
the corresponding firmware doesn't support virtio devices.
However, the current cleanup code doesn't undo the operations
performed by rproc_add() in case the manual boot fails. A
successful rproc_add() requires the corresponding rproc_del()
to be called during cleanup, so fix the same.

Signed-off-by: Suman Anna <s-anna@ti.com>
---
 drivers/remoteproc/pruss_remoteproc.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/remoteproc/pruss_remoteproc.c b/drivers/remoteproc/pruss_remoteproc.c
index da00a80..c528efe 100644
--- a/drivers/remoteproc/pruss_remoteproc.c
+++ b/drivers/remoteproc/pruss_remoteproc.c
@@ -890,7 +890,7 @@ static int pru_rproc_probe(struct platform_device *pdev)
 		ret = rproc_boot(pru->rproc);
 		if (ret) {
 			dev_err(dev, "rproc_boot failed\n");
-			goto put_mbox;
+			goto del_rproc;
 		}
 	}
 
@@ -901,6 +901,8 @@ static int pru_rproc_probe(struct platform_device *pdev)
 
 	return 0;
 
+del_rproc:
+	rproc_del(pru->rproc);
 put_mbox:
 	mbox_free_channel(pru->mbox);
 free_rproc:
-- 
1.7.9.5

